### PR TITLE
build: allow building without openssl, enables native m1 development builds

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
When building on M1, inside of depends, you should use `make NO_PROTOBUF=1 NO_OPENSSL=1 -j8`

### CODE REVIEW

**Review by Munkybooty — DISMISSED** [review]
tACK 

it should be mentioned that you need to run 
```
./configure --prefix=`pwd`/depends/aarch64-apple-darwin21.2.0 --disable-bip70
```

otherwise you'll be met with linker errors:
```
clang: error: linker command failed with exit code 1 (use -v to see invocation)
ld: symbol(s) not found for architecture arm64
```

`make check` finished in 1:55 and functional tests are still sluggish

File: configure.ac [hunk 1]
```diff
@@ -654,12 +654,14 @@ case $host in
...
-         openssl_prefix=`$BREW --prefix openssl 2>/dev/null`
-         bdb_prefix=`$BREW --prefix berkeley-db4 2>/dev/null`
-         qt5_prefix=`$BREW --prefix qt5 2>/dev/null`
-         if test x$openssl_prefix != x; then
-           PKG_CONFIG_PATH="$openssl_prefix/lib/pkgconfig:$PKG_CONFIG_PATH"
-           export PKG_CONFIG_PATH
+         if test x$enable_bip70 != xno; then
+           openssl_prefix=`$BREW --prefix openssl 2>/dev/null`
+           bdb_prefix=`$BREW --prefix berkeley-db4 2>/dev/null`
+           qt5_prefix=`$BREW --prefix qt5 2>/dev/null`
```
Comment (UdjinM6) [inline configure.ac:660]: this doesn't look right imo, should be more like
```suggestion
         bdb_prefix=`$BREW --prefix berkeley-db4 2>/dev/null`
         qt5_prefix=`$BREW --prefix qt5 2>/dev/null`
         if test x$enable_bip70 != xno; then
           openssl_prefix=`$BREW --prefix openssl 2>/dev/null`
```

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: configure.ac (hunk 1 above)
Comment (PastaPastaPasta) [reply configure.ac]: applied the commit, seems to work well

**Comment by ogabrielides** [comment]
Compilation works on m1. 
make check successful along with functional tests.

**Review by ogabrielides — APPROVED** [review]

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
It makes sense to disable bip70 when there is no openssl (cause it's a dependency) but it feels weird to disable openssl when we simply don't want bip70 imo. How about smth like e308bef15b7214572ccac5a32868173b435ef251?

**Comment by Munkybooty** [comment]
OpenSSL unfortunately is the main blocker for Native M1 building. I believed when I had #4612 open that we may finally be able to build on M1 because protobuf being optional gave me the assumption that without bip70 we would have no issue. We realized upon building that that's when the OpenSSL was the true blocker. We use version 1.0.1 and M1 support exists in 1.0.2 and 1.1.1. When @kittywhiskers tried to bump the version we had some nontrivial errors come up iirc. That mostly is the motivation behind disabling it

**Comment by UdjinM6** [comment]
@Munkybooty yes, I get it. My point is that disabling bip70 should not disable openssl cause it doesn't make sense in general (and esp. for non-M1 systems). With the patch I'm proposing you can either build depends with `NO_OPENSSL=1` or specify `--without-openssl` in configure - both options will result in openssl AND bip70 disabled.

**Comment by Munkybooty** [comment]
@UdjinM6 Very fair point, I am in favor of your patch, it makes the most sense logically, I was hoping to clear any confusion if there was any. I see what you mean though

**Comment by PastaPastaPasta** [comment]
What is the use case for including openssl if you're not including bip70? @UdjinM6

**Review by UdjinM6 — APPROVED** [review]
utACK

**Review by Munkybooty — COMMENTED** [review]
post tACK

**Pull request merged**